src/components/Dashboard/MainInput/MainInput.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {
+  InputGroup,
+  Button,
+  FormControl,
+  Row,
+  Container,
+} from 'react-bootstrap';
+
+export default class MainInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newTaskVal: '',
+    };
+    this.updateInputValue = this.updateInputValue.bind(this);
```

Comment:
there's no real use to bind these methods if you use arrow functions in `onclick`
https://reactjs.org/docs/faq-functions.html